meteor-bundle-main.js

```diff
@@ -1,12 +1,119 @@
-// This file is used by Sandstorm to monkey-patch certain classes/functions in Nodejs
-// Specifically, we're changing http.Server.listen to listen on fd=3 instead of the normal
-// parameters. This is fine to do globally, since there's only ever one http server in meteor.
+// Typically, Meteor binds to one port and speaks HTTP.
+//
+// For Sandstorm, `run-bundle` has already bound a number of TCP ports
+// for us and passed them in as file descriptors starting at #3. So
+// this file runs before Meteor starts, monkey-patching the node world
+// so that, along with pre-meteor.js, we create the the right
+// HTTP+HTTPS services.
+
+var fs = require('fs');
 var http = require('http');
+var https = require('https');
 var net = require('net');
+var url = require('url');
+
+function sandstorm_main() {
+  monkeypatchHttpAndHttps();
+
+  // Delegate to Meteor.
+  require("./main.js");
+}
+
+function monkeypatchHttpAndHttps() {
+  // Monkey-patch the HTTP object's listen() function so if Meteor
+  // calls it, then we listen on FD #3.
+  var oldListen = http.Server.prototype.listen;
+  http.Server.prototype.listen = function (port, host, cb) {
+    // Overridable by passing e.g. {fd: 4} as port.
+    if (typeof port == 'object') {
+      return oldListen.call(this, port, host, cb);
+    }
+    return oldListen.call(this, {fd: 3}, cb);
+  }
+
+  // When Meteor calls createServer(), if we are in HTTPS mode, give it a HTTPS server.
+  //
+  // Passing a second argument of `true` allows us to call the real
```

Comment:
What if Node adds a second argument at some point? Any value passed for it would probably appear truthy.

An alternative would be to add `http.createServerForSandstorm` which is called preferentially when available, or maybe at least use the word "sandstorm" as the second parameter so that we can detect unexpected signature changes.